### ISSUE TITLE
Release Planning: Grin v3.0.0

### ISSUE THREAD

**Issue by lehnberg** [mimblewimble/grin-pm#204]
Planning issue for version 3.0.0 of `grin` and `grin-wallet`, which is to support the [scheduled network-wide upgrade](https://www.grin-forum.org/t/grin-v3-0-0-hard-fork-upgrade-jan-2020/6036) occurring at block 524,160, around Jan 15, 2020.

#### Scope freeze
As per the October 15 development meeting, deadline for scope is **17:00 [UTC](http://www.timebie.com/std/utc.php) on October 29**.

#### Target release date
Beta: Early December
Release candidate: Late December
Final release: Early January

#### Discussion
Planning discussion can be had asynchronously on this issue, or in development sub-teams on [keybase](https://keybase.io):
* Install the keybase client and create an account,
* In the client, join team and
   * for node development, enter `grincoin.teams.node_dev`
   * for wallet development, enter `grincoin.teams.wallet_dev`

#### Definitions

**P1 - Critical
P2 - Important
P3 - Fix if time**

🛠: in progress
✅: merged
🔍: awaiting review

---
## Milestones
https://github.com/mimblewimble/grin/milestone/15
https://github.com/mimblewimble/grin-wallet/milestone/6

## Items

Make suggestions by commenting in issue, preferably with a motivation and a proposed priority.

| #  |  Priority | Team | Description | Reference | Status |
|---|---|:---|:---|:---|---|
1 | P1 | Node | Cuckaroo tweak | https://github.com/mimblewimble/grin/pull/3136 | ✅|
2 | P1 | Node | Enable faster sync: Will facilitate some reasonably significant improvements to the initial "fast sync" process | mimblewimble/grin-rfcs#29 | ✅|
3 | P1 | Wallet | TOR RFC and matching functionality | mimblewimble/grin-wallet#226 |✅|
4 | P1 | Wallet | Payment Proofs (blocked by TOR?) |  mimblewimble/grin-wallet#230 | ✅|
5 | P2 | Wallet | Transaction Update Rework | mimblewimble/grin-wallet#244 (needs RFC)? | ✅|
6 | P2 | Node | Node API V2 | mimblewimble/grin-rfcs#28 | ✅ |
7 | P2 | Node | Read header_head and sync_head from header MMRs directly | https://github.com/mimblewimble/grin/pull/3045 | ✅ |
8 | P3 | Node | Refactor transaction building combinators |  https://github.com/mimblewimble/grin/pull/3057 | ✅ |
10 | P3 | Node | TUI Logs view | https://github.com/mimblewimble/grin/pull/3064 | ✅|
11 | P3 | Node | fix: split state validation status into kernel and rproof updates | https://github.com/mimblewimble/grin/pull/3096 | ✅| 
12 | P3 | Wallet | "data_dir" option has been broken | https://github.com/mimblewimble/grin-wallet/issues/251 | ✅ | 

## Out of scope

| # |  Priority | Team | Description | Reference | Motivation
|---|---|:---|:---|:---|:---|
? |P? | Wallet | Interactive CLI | | time constraints|
9 | P3 | Wallet | File locks for wallet (needs definition, but might be very desired for Transaction update rework as well) | mimblewimble/grin-wallet#148 |"won't be happening just now, it was something I thought might be needed but won't be for the time being"|

**Comment by yeastplume** [comment]
#### Wallet P1
* TOR RFC and matching functionality - https://github.com/mimblewimble/grin-wallet/pull/226
* Payment Proofs (which will use TOR addresses/keys but also work without TOR) - https://github.com/mimblewimble/grin-wallet/issues/230

#### P2 
* Transaction Update Rework -https://github.com/mimblewimble/grin-wallet/issues/244 (Will likely need RFC)

#### P3
* File locks for wallet (needs definition, but might be very desired for Transaction update rework as well) https://github.com/mimblewimble/grin-wallet/issues/148

Out of Scope
* Interactive CLI for wallet mentioned earlier, justification being there won't be enough time, what we have there in P1/P2 issues is plenty.

**Comment by antiochp** [comment]
P1 "Enable faster sync" This is a change we would like to get into 3.0.0 scheduled HF as this will facilitate some reasonably significant improvements to the initial "fast sync" process - aka "faster sync". https://github.com/mimblewimble/grin-rfcs/pull/29

**Comment by quentinlesceller** [comment]
P2 - Node API V2 https://github.com/mimblewimble/grin-rfcs/pull/28
